test/integration/allocations_test.rb

```diff
@@ -0,0 +1,35 @@
+require File.dirname(__FILE__) + '/../test_helper'
+
+class CreateAllocationsTest < Test::Unit::TestCase
+  include TestHelper
+
+  def setup
+    @gateway = XeroGateway::Gateway.new(CONSUMER_KEY, CONSUMER_SECRET)
+
+    if STUB_XERO_CALLS
+      @gateway.xero_url = 'DUMMY_URL'
+      @gateway.stubs(:http_put).with { |_client, url, _body, _params| url =~ /CreditNotes\/.*\/Allocations$/ }.returns(get_file_as_string('allocations.xml'))
+    end
+  end
+
+  def test_create_allocation
+    example_allocation = dummy_allocation.dup
```

Comment:
I'm not 100% following why you're `#dup`-ing these sorry?

Reply:
.. actualyl nor am I now ;)